components/client/src/main/java/com/hotels/styx/client/OriginRestrictionLoadBalancingStrategy.java

```diff
@@ -88,7 +86,7 @@ public OriginRestrictionLoadBalancingStrategy(ActiveOrigins activeOrigins, LoadB
     }
 
     private Stream<String> regularExpressionStream(String cookieValue) {
-        return stream(COOKIE_SPLITTER.split(cookieValue).spliterator(), false);
+        return Arrays.stream(cookieValue.split(",")).map(String::trim);
```

Comment:
Depending on where this is used, splitting using a regular expression might be too expensive. Maybe use `StringTokenizer`.

Reply:
Maybe so. It's a bit more fiddly to shoehorn it into a stream, but doable.

Reply:
This code is in the critical path. It would be called once for each HTTP request, subject to configuration.